src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1639,6 +1648,13 @@ def test(
             keep_model = self.net
             self.net = self.ema_model.ema
 
+        context = PhaseContext(
```

Comment:
I think it makes more sense to populate the context after prep_for_test.
I think it would also make sense to add relevant attributes to PhaseContext since we didn't have this phase in the past (like self.test_metrics)